sdk/openai/Azure.AI.OpenAI/src/Custom/AzureOpenAIModelFactory.cs

```diff
@@ -7,29 +7,192 @@
         /// <returns> A new <see cref="OpenAI.Choice"/> instance for mocking. </returns>
-        public static Choice Choice(string text = null, int index = default, CompletionsLogProbabilityModel logProbabilityModel = null, CompletionsFinishReason finishReason = default)
+        public static Choice Choice(
+            string text = null,
+            int index = default,
+            ContentFilterResults contentFilterResults = null,
+            CompletionsLogProbabilityModel logProbabilityModel = null,
+            CompletionsFinishReason finishReason = default)
         {
-            if (text == null)
-            {
-                throw new ArgumentNullException(nameof(text));
-            }
-            // Custom code: remove inappropriate null check of nullable CompletionsLogProbabilityModel
+            return new Choice(text, index, contentFilterResults, logProbabilityModel, finishReason);
+        }
+
+        /// <summary> Initializes a new instance of ChatChoice. </summary>
+        /// <param name="message"> The chat message associated with this chat completions choice </param>
+        /// <param name="index"> The ordered index associated with this chat completions choice. </param>
+        /// <param name="finishReason"> The reason that this chat completions choice completed its generated. </param>
+        /// <param name="deltaMessage"> For streamed choices, the internal representation of a 'delta' payload </param>
+        /// <param name="contentFilterResults"> The category annotations for this chat choice's content filtering </param>
+        /// <returns> A new <see cref="OpenAI.ChatChoice"/> instance for mocking. </returns>
+        public static ChatChoice ChatChoice(
+            ChatMessage message = null,
+            int index = default,
+            CompletionsFinishReason finishReason = default,
+            ChatMessage deltaMessage = null,
+            ContentFilterResults contentFilterResults = null)
+        {
+            return new ChatChoice(message, index, finishReason, deltaMessage, contentFilterResults);
+        }
+
+        /// <summary> Initializes a new instance of ChatCompletions. </summary>
+        /// <param name="id"> A unique identifier associated with this chat completions response. </param>
+        /// <param name="created"> The first timestamp associated with generation activity for this completions response. </param>
+        /// <param name="choices"> The collection of completions choices associated with this completions response. </param>
+        /// <param name="promptFilterResults"> The filter annotations for this chat completions messages input </param>
+        /// <param name="usage"> Usage information for tokens processed and generated as part of this completions operation. </param>
+        /// <returns> A new <see cref="OpenAI.ChatCompletions"/> instance for mocking. </returns>
+        public static ChatCompletions ChatCompletions(
```

Comment:
This one (ChatCompletions) is also already present in the auto-generated one.

Reply:
All part of that naming difference issue -- will answer on the converged comment (glad we caught this!)